lib/src/connected_device_operation.dart

```diff
@@ -45,7 +45,7 @@ class ConnectedDeviceOperationImpl implements ConnectedDeviceOperation {
   Future<List<int>> readCharacteristic(QualifiedCharacteristic characteristic) {
     final specificCharacteristicValueStream = characteristicValueStream
         .where((update) => update.characteristic == characteristic)
-        .map((update) => update.result.dematerialize());
+        .map((update) => update.result.dematerialize()!);
```

Comment:
Is this `result` of type `Result<X?>`? If it's just `Result<T>`, then `.dematerialize()` should return a non-optional type, so that force-unwrapping is not needed.

Please, check this.